Resources/Prototypes/Entities/Objects/Consumable/Food/snacks.yml

```diff
@@ -272,6 +272,46 @@
   - type: Food
     trash: FoodPacketSyndiTrash
 
+- type: entity
+  parent: FoodSnackBase
+  id: DrinkRamen
+  name: cup ramen
+  description: A cheap food with a taste that reminds you of your school years.
+  components:
+  - type: RefillableSolution
+    solution: food
+  - type: InjectableSolution
+    solution: food
+  - type: SolutionContainerManager
+    solutions:
+      food:
+        maxVol: 50 #big cup
+        reagents:
+        - ReagentId: DryRamen
+          Quantity: 30
+        - ReagentId: Soysauce
+          Quantity: 5
+  - type: Sprite
+    state: ramen
+  - type: Food
+    trash: FoodPacketCupRamenTrash
```

Comment:
It should be possible to refill it again after. It makes no sense that the cup loses its ability to hold reagents after you consume it all.

Reply:
I havent found a way to make utensils usable on containers such as cups so i make the cup ramen a snack. is there a way for the item to not delete itself after all the reagents are consumed?

Reply:
> It should be possible to refill it again after. It makes no sense that the cup loses its ability to hold reagents after you consume it all.

Personally I don't think players will be holding on to empty ramen cups, and also you couldn't refill it with ramen anyway.